### PR TITLE
feat(android,ios,js): add getBoundingClientRect api

### PR DESCRIPTION
**Pull request by wwwcg**
Before submitting a new pull request, please make sure:

- [x] Test cases have been added/updated/passed for the code you will submit.
- [x] Documentation has added or updated.
- [x] Commit message is following the [Convention Commit](https://conventionalcommits.org/) guideline with maximum 72 characters.
- [x] Squash the repeat code commits, short patches are welcome.

### CODE REVIEW

**Review by zoomchan-cxj — DISMISSED** [review]
LGTM

**Review by ozonelmy — DISMISSED** [review]
LGTM

File: renderer/native/android/src/main/java/com/tencent/mtt/hippy/uimanager/HippyViewController.java [hunk 1]
```diff
@@ -452,4 +460,47 @@ private void measureInWindow(@NonNull View view, @NonNull Promise promise) {
         result.put("statusBarHeight", PixelUtil.px2dp(statusBarHeight));
         promise.resolve(result);
     }
+
+    private void getBoundingClientRect(@NonNull View view, @NonNull List<?> params,
+            @NonNull Promise promise) {
+        boolean relToContainer = false;
+        if (!params.isEmpty()) {
+            Object param = params.get(0);
```
Comment (siguangli) [inline renderer/native/android/src/main/java/com/tencent/mtt/hippy/uimanager/HippyViewController.java:468]: 统一使用ArrayUtils获取

File: renderer/native/android/src/main/java/com/tencent/mtt/hippy/uimanager/HippyViewController.java [hunk 2]
```diff
@@ -452,4 +460,47 @@ private void measureInWindow(@NonNull View view, @NonNull Promise promise) {
...
+            if (param instanceof HashMap) {
+                relToContainer = ((HashMap<?, ?>) param).get(KEY_REL_TO_CONTAINER) == Boolean.TRUE;
```
Comment (siguangli) [inline renderer/native/android/src/main/java/com/tencent/mtt/hippy/uimanager/HippyViewController.java:470]: 统一使用MapUtils来获取

File: renderer/native/android/src/main/java/com/tencent/mtt/hippy/uimanager/HippyViewController.java (hunk 1 above)
Comment (iPel) [reply renderer/native/android/src/main/java/com/tencent/mtt/hippy/uimanager/HippyViewController.java]: 已修改

File: renderer/native/android/src/main/java/com/tencent/mtt/hippy/uimanager/HippyViewController.java (hunk 2 above)
Comment (iPel) [reply renderer/native/android/src/main/java/com/tencent/mtt/hippy/uimanager/HippyViewController.java]: 已修改

**Review by ozonelmy — APPROVED** [review]
LGTM

**Review by zoomchan-cxj — DISMISSED** [review]

File: renderer/native/android/src/main/java/com/tencent/mtt/hippy/uimanager/HippyViewController.java [hunk 3]
```diff
@@ -452,4 +462,47 @@ private void measureInWindow(@NonNull View view, @NonNull Promise promise) {
...
+            Map<String, Object> param = ArrayUtils.getMapValue(params, 0);
+            if (param != null) {
+                relToContainer = MapUtils.getBooleanValue(param, KEY_REL_TO_CONTAINER, false);
+            }
+        }
+        int x;
+        int y;
+        int width = view.getWidth();
+        int height = view.getHeight();
+        int[] pair = new int[2];
```
Comment (siguangli) [inline renderer/native/android/src/main/java/com/tencent/mtt/hippy/uimanager/HippyViewController.java:479]: pair的定义最好放在下面 view.getLocationInWindow(pair); 之前，不然走rootview == null分支return以后就白new一个对象

**Review by siguangli — CHANGES_REQUESTED** [review]

**Review by ozonelmy — DISMISSED** [review]
LGTM

File: renderer/native/android/src/main/java/com/tencent/mtt/hippy/uimanager/HippyViewController.java (hunk 3 above)
Comment (iPel) [reply renderer/native/android/src/main/java/com/tencent/mtt/hippy/uimanager/HippyViewController.java]: 已修改

**Review by zoomchan-cxj — APPROVED** [review]

**Review by ozonelmy — APPROVED** [review]
LGTM

**Review by siguangli — APPROVED** [review]

**Pull request merged**